frontend/src/pages/integrations/databricks/authorize.tsx

```diff
@@ -0,0 +1,118 @@
+import { useState } from "react";
+import Head from "next/head";
+import Image from "next/image";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import { faArrowUpRightFromSquare, faBookOpen } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import { useSaveIntegrationAccessToken } from "@app/hooks/api";
+
+import { Button, Card, CardTitle, FormControl, Input } from "../../../components/v2";
+
+export default function DatabricksCreateIntegrationPage() {
+  const router = useRouter();
+  const { mutateAsync } = useSaveIntegrationAccessToken();
+
+  const [apiKey, setApiKey] = useState("");
+  const [instanceURL, setInstanceURL] = useState("");
+  const [apiKeyErrorText, setApiKeyErrorText] = useState("");
+  const [instanceURLErrorText, setInstanceURLErrorText] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleButtonClick = async () => {
+    try {
+      setApiKeyErrorText("");
+      setInstanceURLErrorText("");
+      if (apiKey.length === 0) {
+        setApiKeyErrorText("API Key cannot be blank");
+        return;
+      }
+      if (instanceURL.length === 0) {
+        setInstanceURLErrorText("Instance URL cannot be blank");
+        return;
+      }
+
+      setIsLoading(true);
+
+      const integrationAuth = await mutateAsync({
+        workspaceId: localStorage.getItem("projectData.id"),
+        integration: "databricks",
+        url: instanceURL,
+        accessToken: apiKey
+      });
+
+      setIsLoading(false);
+
+      router.push(`/integrations/databricks/create?integrationAuthId=${integrationAuth.id}`);
+    } catch (err) {
+      console.error(err);
+    }
```

Comment:
`setIsLoading(false);` should probably be moved after the `catch` or in a `finally` so that it is set `false` whether it fails or succeeds. Alternatively use `isLoading` from mutation hook above to not have to manage this

Reply:
But shouldn't it be before the router.push though?

Reply:
Setting it before or after router.push doesn't matter in this case, because router.push is fast and will finish before the page reloads anyways, I would move `setIsLoading(false)` to the `finally` block like Scott mentioned